### ISSUE TITLE
Merge SLPK files

### ISSUE THREAD

**Issue by xiaoxiong314** [Esri/i3s-spec#101]
I have several Point slpk files, which have different attribute tables. Can I merge these this slpk files into one?

**Comment by rvargasESRI** [comment]
Currently there is no merging mechanism for SLPKs, regardless different or similar attribute tables.  
Do you have access to the original data?  If so, you would be able to regenerate the SLPK.

**Comment by xiaoxiong314** [comment]
Yes, I have access to the original data. I create these point feature classes in ArcGIS Pro. But I don't know how to export them into one SLPK. The "Input Dataset" of the tool-"Create Point Scene Layer Package" allows only one feature layer.
![image](https://user-images.githubusercontent.com/18329996/93404947-354dfe80-f8be-11ea-935b-24562b80854b.png)
![image](https://user-images.githubusercontent.com/18329996/93404974-4d258280-f8be-11ea-86ac-1827c6642bd9.png)

**Comment by rvargasESRI** [comment]
@xiaoxiong314 Could you please send a question at https://community.esri.com/community/gis/applications/arcgis-pro who could better assist you with this request.

**Comment by Biboba** [comment]
Following your excellent blog post (and python script!), it seems doable:
https://www.esri.com/arcgis-blog/products/arcgisrealitystudio/3d-gis/case-study-how-to-maintain-a-statewide-mesh-for-a-digital-twin

I was able to run it successfully with 2 .slpks ! Very exciting (been waiting to do so for a while) ! Just got one 500 for a specific nodepages that I need to investigate but it's very promising !

I think it would be really awesome to include this .slpk merging functionnality to the "i3s_converter.exe" tool and eventually as an ArcGIS Pro GP later on !

**Comment by Ivonne-S** [comment]
Glad you were able to use the script for appending data. We will consider merging of SLPK in the future. Because of the requirements for features based scene layers will need to make sure that cases with duplicated OID and different attributes will work.

**Comment by CJHacking93** [comment]
@Biboba, how did you convert the appended ESLPK containing the multiple ESLPKs (extracted via the i3s-converter.exe tool) created with the python script back to a SLPK which can then be used? Or is my approach incorrect?   

These are the steps I have followed, based on the Blog you posted:

1. Each SLPK extent has been buffered by -50m.
2. Each extent has then been clipped by this negative buffer, yielding smaller extents without the edge distortion from the Site Scan processing. 
3. These areas still overlap, and have some distortion within them, so trying to determine the best solution to merge them into a single layer.
4. I have then used the i3s-converter.exe file to convert from SLPK to ESLPK files successfully (or so I believe to be successful).
5. I have then used the [eslpk_merge.py](https://3dcities.maps.arcgis.com/home/item.html?id=c0dc99aee094424a9b77f17ca932ff3f) (from @Ivonne-S) file to merge multiple (for testing purposes just two) ESLPK files into a single ESLPK file (Again, the script suceeded, but I cannot confirm success or not).
6. Now I am stuck on how to convert this back to a useable format? Potentially I am missing vital steps along the way?

**Comment by Biboba** [comment]
Hi @Hackster93,

I used 7zip with no compression to compress all the folder of the .eslpk :

![Image](https://github.com/user-attachments/assets/2b727d37-ec8f-432a-9043-b259c04cbe82)

> An SLPK is a [zip](https://en.wikipedia.org/wiki/Zip_(file_format)) archive containing compressed files and resources. The compression level for a SLPK file is not compressed. For example, if using 7-zip to create a scene layer package the compression level is `STORE`. The individual resources within the SLPK may be compressed using the compression method `GZIP`. For example, `.json.gz`. The exception is for jpg and png files.  We recommend compressing all resources.  

[Reference I3S Spec](https://github.com/Esri/i3s-spec/blob/7bd83b3143c68bb7c77986ba571517222e8a636d/format/Indexed%203d%20Scene%20Layer%20Format%20Specification.md?plain=1#L237)

Personnally, as I mentionned in my previous post, on my side this workflow did not fully worked: at the end I ended up having a functional merged webscene layer but some tiles did not load properly with 500 errors which results in a coarse mesh in some areas. I never found the solution... Hope there will be a more standard way of achieving "slpk" merging.

Curious to know if you observe the same phenomena on your side.
